bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/wrappers/classic/BigtableHBaseClassicSettings.java

```diff
@@ -15,101 +15,22 @@
-import static com.google.cloud.bigtable.hbase.BigtableOptionsFactory.MAX_INFLIGHT_RPCS_KEY;
-import static com.google.cloud.bigtable.hbase.BigtableOptionsFactory.MAX_SCAN_TIMEOUT_RETRIES;
-import static com.google.cloud.bigtable.hbase.BigtableOptionsFactory.READ_BUFFER_SIZE;
-import static com.google.cloud.bigtable.hbase.BigtableOptionsFactory.READ_PARTIAL_ROW_TIMEOUT_MS;
-import static com.google.common.base.Strings.isNullOrEmpty;
-
 import com.google.api.core.InternalApi;
-import com.google.auth.Credentials;
 import com.google.cloud.bigtable.config.BigtableOptions;
-import com.google.cloud.bigtable.config.BulkOptions;
-import com.google.cloud.bigtable.config.CallOptionsConfig;
-import com.google.cloud.bigtable.config.CredentialOptions;
-import com.google.cloud.bigtable.config.RetryOptions;
-import com.google.cloud.bigtable.hbase.BigtableExtendedConfiguration;
 import com.google.cloud.bigtable.hbase.BigtableOptionsFactory;
 import com.google.cloud.bigtable.hbase.wrappers.BigtableHBaseSettings;
-import com.google.common.base.Preconditions;
-import io.grpc.Status;
-import java.io.FileInputStream;
-import java.io.FileNotFoundException;
 import java.io.IOException;
 import org.apache.hadoop.conf.Configuration;
-import org.apache.hadoop.hbase.util.VersionInfo;
 
 /** For internal use only - public for technical reasons. */
 @InternalApi("For internal usage only")
 public class BigtableHBaseClassicSettings extends BigtableHBaseSettings {
 
-  private final Configuration configuration;
   private final BigtableOptions bigtableOptions;
 
   public BigtableHBaseClassicSettings(Configuration configuration) throws IOException {
     super(configuration);
-    // we can't create a defensive copy because it might be an instance
-    // BigtableExtendedConfiguration
-    this.configuration = configuration;
-    BigtableOptions.Builder bigtableOptionsBuilder = BigtableOptions.builder();
-
-    bigtableOptionsBuilder.setProjectId(getProjectId()).setInstanceId(getInstanceId());
-
-    String appProfileId = configuration.get(APP_PROFILE_ID_KEY);
-    if (!isNullOrEmpty(appProfileId)) {
-      bigtableOptionsBuilder.setAppProfileId(appProfileId);
-    }
-
-    configureBulkOptions(bigtableOptionsBuilder);
-    configureChannelOptions(bigtableOptionsBuilder);
-    configureClientCallOptions(bigtableOptionsBuilder);
-    configureCredentialOptions(bigtableOptionsBuilder);
-    configureRetryOptions(bigtableOptionsBuilder);
-
-    String emulatorHost = configuration.get(BigtableOptionsFactory.BIGTABLE_EMULATOR_HOST_KEY);
-    if (!isNullOrEmpty(emulatorHost)) {
-      bigtableOptionsBuilder.enableEmulator(emulatorHost);
-    }
-
-    String useBatchStr = configuration.get(BIGTABLE_USE_BATCH);
-    if (!isNullOrEmpty(useBatchStr)) {
-      bigtableOptionsBuilder.setUseBatch(Boolean.parseBoolean(useBatchStr));
-    }
-
-    this.bigtableOptions = bigtableOptionsBuilder.build();
+    this.bigtableOptions = BigtableOptionsFactory.fromConfiguration(configuration);
```

Comment:
confirming this is already tested somewhere?

Reply:
There is `TestBigtableOptionsFactory`, which tests some of this stuff and it could be improved. But expanding that test is out of scope for this PR especially since we are deleting BIgtableHBaseClassicSettings shortly after the alpha